### PR TITLE
add doc for /users/{user_id}/verify/email

### PR DESCRIPTION
**Pull request by cpanato**
PR for the implementation: https://github.com/mattermost/platform/pull/5502

### CODE REVIEW

File: source/v4/users.yaml [hunk 1]
```diff
@@ -581,3 +581,39 @@
           $ref: '#/responses/Unauthorized'
         '403':
           $ref: '#/responses/Forbidden'
+
+  /users/{user_id}/verify/email:
+    post:
+      tags:
+        - users
+      summary: Verify User sign-up
```
Comment (jwilander) [inline source/v4/users.yaml]: Can we make this `Verify user email`?

File: source/v4/users.yaml [hunk 2]
```diff
@@ -581,3 +581,39 @@
...
+      description: |
+        Verify User account and mark as verified in the database.
```
Comment (jwilander) [inline source/v4/users.yaml]: Can we make this `Verify the email used by a user to sign-up their account with.`?

File: source/v4/users.yaml [hunk 3]
```diff
@@ -581,3 +581,39 @@
...
+            properties:
+              userId:
+                description: The User GUID
+                type: string
+              hashId:
+                description: The hash with UserId and InviteSaltId
+                type: string
+      responses:
+        '200':
+          description: User verified successful
```
Comment (jwilander) [inline source/v4/users.yaml]: And this to `User verification successful`?

File: source/v4/users.yaml (hunk 1 above)
Comment (cpanato) [inline source/v4/users.yaml]: done

File: source/v4/users.yaml (hunk 2 above)
Comment (cpanato) [inline source/v4/users.yaml]: done

File: source/v4/users.yaml (hunk 3 above)
Comment (cpanato) [inline source/v4/users.yaml]: done

**Comment by jwilander** [comment]
@cpanato can you give this a rebase and then I can merge it?

**Comment by cpanato** [comment]
@jwilander sure, i will do it tomorrow I forgot my yubico at home and then I cannot commit :/

**Comment by jwilander** [comment]
No rush, thanks @cpanato

**Comment by cpanato** [comment]
done @jwilander

**Pull request merged**